### PR TITLE
soft-delete-cost-analysis: fixing return type

### PR DESCRIPTION
**Pull request by marcinlimanski**
## Description
Fixing run time error with return type

### CODE REVIEW

File: storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def soft_delete_relative_cost_analyzer(
     agg_days: int = 30,
     lookback_days: int = 360,
     list_buckets: bool = False,
-) -> str | Dict[str, float]:  # Note potential string output
+    ) -> Union[str, Dict[str, float]]: # Note potential string output
```
Comment (msampathkumar) [inline storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py:241]: Thanks for the PR. As per https://docs.python.org/3/library/typing.html#special-forms, `Union` does the same job as `|`. Any particular reason why change this ?

Comment (marcinlimanski) [reply storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py:241]: Thanks for quick replay! Running the script with python 3.9 I got this issue:

python-docs-samples/storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py", line 241, in <module>
    ) -> str | Dict[str, float]:  # Note potential string output
TypeError: unsupported operand type(s) for |: 'type' and '_GenericAlias'

The above change fixes this.

Comment (marcinlimanski) [reply storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py:241]: The shorthand '|' for union is only supported in python 3.10 and up.

**Comment by iennae** [comment]
@marcinlimanski thanks for the PR and your first time contribution to the repository. It looks like you need to sign the CLA before we can accept this request.

**Comment by marcinlimanski** [comment]
@iennae Thanks for the reminder, CLA is signed. PR ready to go.

File: storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py (hunk 1 above)
Comment (davidcavazos) [reply storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py:241]: We can still use `|` in Python 3.9 by adding `from __future__ import annotations`, just confirmed in my local setup.

File: storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py [hunk 2]
```diff
@@ -31,7 +31,7 @@
 
 import argparse
 import json
-from typing import Dict, List
+from typing import Dict, List, Union
```
Comment (davidcavazos) [inline storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py:34]: As per [PEP-585](https://peps.python.org/pep-0585/), we can use `dict` and `list` directly (like `dict[str, float]`), so we can get rid of this whole import.

File: storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py [hunk 3]
```diff
@@ -238,7 +236,7 @@ def soft_delete_relative_cost_analyzer(
     agg_days: int = 30,
     lookback_days: int = 360,
     list_buckets: bool = False,
-) -> str | Dict[str, float]:  # Note potential string output
+    ) -> str | Dict[str, float]: # Note potential string output
```
Comment (davidcavazos) [inline storage/cost-analysis/storage_soft_delete_relative_cost_analyzer.py:239]: `Dict[str, float]` -> `dict[str, float]`

**Review by davidcavazos — APPROVED** [review]

**Pull request merged**